gp/cartesian_graph.py

```diff
@@ -105,22 +105,22 @@ def parse_genome(self, genome):
 
         self._determine_active_nodes()
 
-    def _hidden_column_idx(self, idx):
+    def _hidden_column_idx(self, idx: int) -> int:
         return (idx - self._n_inputs) // self._n_rows
 
     @property
-    def input_nodes(self):
-        return self._nodes[: self._n_inputs]
+    def input_nodes(self) -> List[gp.Node]:
+        return self._nodes[:self._n_inputs]
 
     @property
-    def hidden_nodes(self):
-        return self._nodes[self._n_inputs : -self._n_outputs]
+    def hidden_nodes(self) -> List[gp.Node]:
+        return self._nodes[self._n_inputs:-self._n_outputs]
 
     @property
-    def output_nodes(self):
-        return self._nodes[-self._n_outputs :]
+    def output_nodes(self) -> List[gp.Node]:
+        return self._nodes[-self._n_outputs:]
 
-    def _determine_active_nodes(self):
+    def _determine_active_nodes(self) -> dict:
```

Comment:
is it possible to indicate dict of sets like `Dict[Set]`?

Reply:
Yep, changed this.